### PR TITLE
feat: Added API for getting historical data for a token

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
A short summary which serves as a squashed-commit message.

A description to understand introduced changes without reading the code.

Important changes:
- [x] Something worth noting for reviewers.

Closes #

### CODE REVIEW

File: services/wallet/cryptocompare.go [hunk 1]
```diff
@@ -32,6 +32,22 @@ type MarketCoinValues struct {
...
+
+type HistoricalValuesContainer struct {
+	Aggregated     bool                   `json:"Aggregated"`
+	TimeFrom       int64                  `json:"TimeFrom"`
+	TimeTo         int64                  `json:"TimeTo"`
+	HistoricalData []TokenHistoricalPairs `json:"Data"`
+}
+
+type HistoricalValuesData struct {
+	Data HistoricalValuesContainer `json:"Data"`
```
Comment (alaibe) [inline services/wallet/cryptocompare.go:52]: why do u need a struct containing only one attribute?

**Review by alaibe — APPROVED** [review]

File: services/wallet/cryptocompare.go [hunk 2]
```diff
@@ -127,3 +145,57 @@ func fetchTokenMarketValues(symbols []string, currency string) (map[string]Marke
 	return item, nil
 
 }
+
+func fetchHourlyMarketValues(symbol string, currency string, limit int, aggregate int) ([]TokenHistoricalPairs, error) {
+	item := []TokenHistoricalPairs{}
+	httpClient := http.Client{Timeout: time.Minute}
+
+	url := fmt.Sprintf("https://min-api.cryptocompare.com/data/v2/histohour?fsym=%s&tsym=%s&aggregate=%d&limit=%d&extraParams=Status.im", symbol, currency, aggregate, limit)
+	resp, err := httpClient.Get(url)
```
Comment (0x-r4bbit) [inline services/wallet/cryptocompare.go:156]: Probably makes sense to add a client timeout here

File: services/wallet/cryptocompare.go (hunk 1 above)
Comment (0x-r4bbit) [reply services/wallet/cryptocompare.go:52]: Looks like this is needed for marshaling. Looks like the the response of the http requests returns Json that has a `Data` field.

File: services/wallet/cryptocompare.go [hunk 3]
```diff
@@ -127,3 +145,57 @@ func fetchTokenMarketValues(symbols []string, currency string) (map[string]Marke
...
+	item = container.Data.HistoricalData
+
+	return item, nil
+}
+
+func fetchDailyMarketValues(symbol string, currency string, limit int, allData bool, aggregate int) ([]TokenHistoricalPairs, error) {
+	item := []TokenHistoricalPairs{}
+	httpClient := http.Client{Timeout: time.Minute}
+
+	url := fmt.Sprintf("https://min-api.cryptocompare.com/data/v2/histoday?fsym=%s&tsym=%s&aggregate=%d&limit=%d&allData=%v&extraParams=Status.im", symbol, currency, aggregate, limit, allData)
```
Comment (0x-r4bbit) [inline services/wallet/cryptocompare.go]: This base URL is used in multiple places now. Probably makes sense to move that into a reusable thing.

**Review by 0x-r4bbit — CHANGES_REQUESTED** [review]

File: services/wallet/cryptocompare.go (hunk 1 above)
Comment (Khushboo-dev-cpp) [reply services/wallet/cryptocompare.go:52]: Exactly! needed it for unmarshalling

File: services/wallet/cryptocompare.go (hunk 2 above)
Comment (Khushboo-dev-cpp) [reply services/wallet/cryptocompare.go:156]: Hey on line 151, its already defined? 	
`httpClient := http.Client{Timeout: time.Minute}`

File: services/wallet/cryptocompare.go (hunk 3 above)
Comment (Khushboo-dev-cpp) [reply services/wallet/cryptocompare.go]: done

**Comment by Khushboo-dev-cpp** [comment]
@PascalPrecht I have addressed al comments :)

**Review by 0x-r4bbit — APPROVED** [review]

**Pull request merged**